pandas/core/frame.py

```diff
@@ -1196,14 +1196,23 @@ def to_records(self, index=True, convert_datetime64=True):
         ----------
         index : boolean, default True
             Include index in resulting record array, stored in 'index' field
-        convert_datetime64 : boolean, default True
+        convert_datetime64 : boolean, default False
+            .. deprecated:: 0.23.0
+
             Whether to convert the index to datetime.datetime if it is a
             DatetimeIndex
 
         Returns
         -------
         y : recarray
         """
+
+        if convert_datetime64:
```

Comment:
is not None. We are going to completely remove this parameter, so we want to warn if its passed at all.